src/commands/dev/lib/builder-cache.ts

```diff
@@ -194,7 +166,9 @@ export async function getBuilder(
   if (!builderWithPkg) {
     const cacheDir = await builderDirPromise;
     const parsed = npa(builderPkg);
-    const dest = join(cacheDir, 'node_modules', parsed.name || builderPkg);
+    const buildersPkg = await readJSON(join(cacheDir, 'package.json'));
+    const pkgName = getPackageName(parsed, buildersPkg) || builderPkg;
```

Comment:
`pkgName` should be string
```suggestion
    const pkgName = getPackageName(parsed, buildersPkg) || builderPkg.name || 'unknown-builder';
```

Reply:
Note that it's `builderPkg`, which is the argument passed in to this function, not `buildersPkg`.